src/js/core/methods/helpers/binanceSignTx.js

```diff
@@ -2,42 +2,33 @@
 
 import { ERRORS } from '../../../constants';
 import { validateParams } from './paramsValidator';
-import type { MessageResponse, DefaultMessageResponse } from '../../../device/DeviceCommands';
-import type {
-    BinanceTxRequest,
-    BinanceSignedTx,
-} from '../../../types/trezor/protobuf';
 import type {
     BinanceSDKTransaction,
     BinancePreparedMessage,
     BinancePreparedTransaction,
 } from '../../../types/networks/binance';
+import type { TypedCall } from '../../../types/trezor/protobuf';
 
-const processTxRequest = async (typedCall: (type: string, resType: string, msg: Object) => Promise<DefaultMessageResponse>,
-    response: MessageResponse<BinanceTxRequest>,
+const processTxRequest = async (
+    typedCall: TypedCall,
     messages: BinancePreparedMessage[],
     index: number,
-): Promise<BinanceSignedTx> => {
-    const msg = messages[index];
-    const type = msg.type;
+) => {
+    const { type, ...params } = messages[index];
+    // const type = msg.type;
     const lastOp = (index + 1 >= messages.length);
 
     if (lastOp) {
-        const response: MessageResponse<BinanceSignedTx> = await typedCall(type, 'BinanceSignedTx', {
-            ...msg,
-            type: null, // 'type' is not a protobuf field and needs to be removed
-        });
+        // $FlowIssue typedCall TEMP
+        const response = await typedCall(type, 'BinanceSignedTx', params);
         return response.message;
     }
-    const ack: MessageResponse<BinanceTxRequest> = await typedCall(type, 'BinanceTxRequest', {
-        ...msg,
-        type: null, // 'type' is not a protobuf field and needs to be removed
```

Comment:
hmm, `type: null` why it was there? some trezor-link oddity? :thinking:

Reply:
yes, there is this union type: string + ProtobufExactMessage. previously "type" was removed because it was not a valid protobuf message when used.
current solution destructure this union therefore type is not present in sent message. but now the problem with the union appeared :)